code/ship/ship.cpp

```diff
@@ -3291,10 +3302,12 @@ int parse_ship_values(ship_info* sip, const bool is_template, const bool first_t
 		Error(LOCATION, "%s '%s' has too many primary banks (%d).  Maximum for ships is currently %d.\n", info_type_name, sip->name, sip->num_primary_banks, MAX_SHIP_PRIMARY_BANKS);
 	}
 
+	memset(sip->allowed_weapons, 0, sizeof(int) * MAX_WEAPON_TYPES);
```

Comment:
I agree with @MageKing17.  I like keeping things consistent, and if someone does decide to change the implementation of `sip->allowed_weapons`, this is one of the most obvious things to look for when refactoring.